lib/administration/ticket/include/irods/ticket_administration.hpp

```diff
@@ -0,0 +1,578 @@
+        ///
+        std::string create_ticket(admin_tag, RxComm& conn, type _type, std::string_view obj_path);
+
+        ///
+        /// \brief Delete the ticket that is specified
+        ///
+        /// \param[in] conn The communication object
+        /// \param[in] ticket_name Name of the ticket that is to be deleted
+        ///
+        void delete_ticket(RxComm& conn, std::string_view ticket_name);
+
+        ///
+        /// \brief Delete the ticket that is specified
+        ///
+        /// \param[in] conn The communication object
+        /// \param[in] ticket_id The ticket ID for the ticket that should be deleted
+        ///
+        void delete_ticket(RxComm& conn, int ticket_id);
+
+        ///
+        /// \brief Delete the ticket that is specified using admin privilige
+        ///
+        /// \param[in] admin_tag Struct tag to indicate admin privilige
+        /// \param[in] conn The communication Object
+        /// \param[in] ticket_name Name of the ticket that is to be deleted
+        ///
+        void delete_ticket(admin_tag, RxComm& conn, std::string_view ticket_name);
+
+        ///
+        /// \brief Delete the ticket that is specified using admin privilige
+        ///
+        /// \param[in] admin_tag Struct tag to indicate admin privilige
+        /// \param[in] conn The communication object
+        /// \param[in] ticket_id The ticket ID for the ticket that should be deleted
+        ///
+        void delete_ticket(admin_tag, RxComm& conn, int ticket_id);
+
+        ///
+        /// \brief Add the ticket constraint to the specified ticket
+        ///
+        /// \tparam ticket_constraint The template struct to indicate the type of constraint (only user, group, and host
+        /// constraint struct) \param[in] conn The communication object \param[in] ticket_name The name of the ticket
+        /// \param[in] constraint The constraint that should be added to the ticket
+        ///
+        template <typename ticket_constraint>
+        void add_ticket_constraint(RxComm& conn,
+                                   const std::string_view ticket_name,
+                                   const ticket_constraint& constraint)
+        {
+            if constexpr (std::is_same_v<user_constraint, ticket_constraint>) {
+                detail::execute_ticket_operation(conn, "mod", ticket_name, "add", "user", constraint.name, "", false);
+            }
+            else if constexpr (std::is_same_v<group_constraint, ticket_constraint>) {
+                detail::execute_ticket_operation(conn, "mod", ticket_name, "add", "group", constraint.name, "", false);
+            }
+            else if constexpr (std::is_same_v<host_constraint, ticket_constraint>) {
+                detail::execute_ticket_operation(conn, "mod", ticket_name, "add", "host", constraint.name, "", false);
+            }
+            else {
+                throw std::invalid_argument("Wrong type object given");
```

Comment:
"Wrong type object given" -> "Unknown constraint type." or "Invalid constraint type." or "Constraint type is not supported."